x/evm/keeper/state_transition.go

```diff
@@ -131,41 +130,37 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 
 	k.SetTxHashTransient(tx.Hash())
 	k.IncreaseTxIndexTransient()
-	res, err := k.ApplyMessage(evm, msg, ethCfg)
+	res, err := k.ApplyMessage(evm, msg, ethCfg, false)
 	if err != nil {
 		return nil, stacktrace.Propagate(err, "failed to apply ethereum core message")
 	}
 
+	rsp := types.TxResponseFromResult(res)
+
 	txHash := tx.Hash()
-	res.Hash = txHash.Hex()
+	rsp.Hash = txHash.Hex()
 	logs := k.GetTxLogs(txHash)
 
 	// Commit and switch to original context
-	if !res.Reverted {
+	if !res.Failed() {
 		commit()
 	}
-	k.ctx = originalCtx
 
 	// Logs needs to be ignored when tx is reverted
 	// Set the log and bloom filter only when the tx is NOT REVERTED
-	if !res.Reverted {
-		res.Logs = types.NewLogsFromEth(logs)
+	if !res.Failed() {
+		rsp.Logs = types.NewLogsFromEth(logs)
 		// Update block bloom filter in the original context because blockbloom is set in EndBlock
 		bloom := k.GetBlockBloomTransient()
 		bloom.Or(bloom, big.NewInt(0).SetBytes(ethtypes.LogsBloom(logs)))
 		k.SetBlockBloomTransient(bloom)
 	}
 
-	// refund gas prior to handling the vm error in order to set the updated gas meter
-	leftoverGas := msg.Gas() - res.GasUsed
-	leftoverGas, err = k.RefundGas(msg, leftoverGas)
-	if err != nil {
-		return nil, stacktrace.Propagate(err, "failed to refund gas leftover gas to sender %s", msg.From())
-	}
 	// update the gas used after refund
-	res.GasUsed = msg.Gas() - leftoverGas
-	k.resetGasMeterAndConsumeGas(res.GasUsed)
-	return res, nil
+	k.ctx = originalCtx
```

Comment:
See my previous PR.

You need to set back to original context before setting up the bloomFiler in transient (k.SetBlockBloomTransient), otherwise it wont be take into account in endblock

Reply:
fixed